src/Storages/System/StorageSystemCertificates.cpp

```diff
@@ -0,0 +1,222 @@
+#include <Common/config.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Storages/System/StorageSystemCertificates.h>
+#include <regex>
+#include <filesystem>
+#include "Poco/File.h"
+#if USE_SSL
+    #include <openssl/x509v3.h>
+    #include "Poco/Net/SSLManager.h"
+    #include "Poco/Crypto/X509Certificate.h"
+#endif
+
+namespace DB
+{
+
+NamesAndTypesList StorageSystemCertificates::getNamesAndTypes()
+{
+    return
+    {
+        {"version",         std::make_shared<DataTypeNumber<Int32>>()},
+        {"serial_number",   std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"signature_algo",  std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"issuer",          std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"not_before",      std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"not_after",       std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"subject",         std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"pkey_algo",       std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"path",            std::make_shared<DataTypeString>()},
+        {"default",         std::make_shared<DataTypeNumber<UInt8>>()}
+    };
+}
+
+#if USE_SSL
+
+static std::unordered_set<std::string> parse_dir(const std::string & dir)
+{
+    std::unordered_set<std::string> ret;
+
+    if (dir.empty())
+        return ret;
+
+    size_t len;
+    const char * ss;
+    const char * s = dir.c_str();
+    const char * p = s;
+    do
+    {
+        if ((*p == ':') || (*p == '\0'))
+        {
+            ss = s;
+            s = p + 1;
+            len = p - ss;
+            if (len == 0)
+                continue;
+            ret.emplace(ss, len);
+        }
+    } while (*p++ != '\0');
+
+    return ret;
+}
+
+static void populateTable(const X509 * cert, MutableColumns & res_columns, const std::string & path, bool def)
+{
+    BIO * b = BIO_new(BIO_s_mem());
+    size_t col = 0;
+
+    res_columns[col++]->insert(X509_get_version(cert) + 1);
+
+    {
+        char buf[1024] = {0};
+        const ASN1_INTEGER * sn = cert->cert_info->serialNumber;
+        BIGNUM * bnsn = ASN1_INTEGER_to_BN(sn, nullptr);
+        if (BN_print(b, bnsn) > 0 && BIO_read(b, buf, sizeof(buf)) > 0)
+            res_columns[col]->insert(buf);
+        else
+            res_columns[col]->insert(Field());
```

Comment:
I guess we can use `insertDefault()` instead of `insert(Field())` here and in the other places

Reply:
done